Source/Core/VideoCommon/PixelShaderManager.cpp

```diff
@@ -14,56 +14,32 @@
 
 bool PixelShaderManager::s_bFogRangeAdjustChanged;
 bool PixelShaderManager::s_bViewPortChanged;
-
-std::array<int4,4> PixelShaderManager::s_tev_color;
-std::array<int4,4> PixelShaderManager::s_tev_konst_color;
+bool PixelShaderManager::s_bEFBScaleChanged;
 
 PixelShaderConstants PixelShaderManager::constants;
 bool PixelShaderManager::dirty;
 
 void PixelShaderManager::Init()
 {
+	s_bFogRangeAdjustChanged = false;
+	s_bViewPortChanged = false;
+	s_bEFBScaleChanged = false;
+
 	memset(&constants, 0, sizeof(constants));
-	memset(s_tev_color.data(), 0, sizeof(s_tev_color));
-	memset(s_tev_konst_color.data(), 0, sizeof(s_tev_konst_color));
 
 	Dirty();
 }
 
 void PixelShaderManager::Dirty()
 {
+	// This function is called after a savestate is loaded.
+	// Any constants that can changed based on settings should be re-calculated
 	s_bFogRangeAdjustChanged = true;
-	s_bViewPortChanged = true;
+	s_bEFBScaleChanged = true;
 
-	for (unsigned index = 0; index < s_tev_color.size(); ++index)
-	{
-		for (int comp = 0; comp < 4; ++comp)
-		{
-			SetTevColor(index, comp, s_tev_color[index][comp]);
-			SetTevKonstColor(index, comp, s_tev_konst_color[index][comp]);
-		}
-	}
-
-	SetAlpha();
-	SetDestAlpha();
-	SetZTextureBias();
-	SetViewportChanged();
-	SetIndTexScaleChanged(false);
-	SetIndTexScaleChanged(true);
-	SetIndMatrixChanged(0);
-	SetIndMatrixChanged(1);
-	SetIndMatrixChanged(2);
-	SetZTextureTypeChanged();
-	SetTexCoordChanged(0);
-	SetTexCoordChanged(1);
-	SetTexCoordChanged(2);
-	SetTexCoordChanged(3);
-	SetTexCoordChanged(4);
-	SetTexCoordChanged(5);
-	SetTexCoordChanged(6);
-	SetTexCoordChanged(7);
-	SetFogColorChanged();
```

Comment:
Also startup. Not all values are initialized to 0 just because of bpmem is zero.